### PR TITLE
feat(settings): allow set user and group for worker and task worker c…

### PR DESCRIPTION
**Pull request by pistej**
…hild processes

### CODE REVIEW

File: src/Server/DefaultHttpServerConfiguration.php [hunk 1]
```diff
@@ -199,6 +205,20 @@ public function getCoroutinesEnabled(): bool
             && $this->settings[self::SWOOLE_HTTP_SERVER_CONFIG_ENABLE_COROUTINE];
     }
 
+    public function getUser(): string
+    {
+        return $this->settings[self::SWOOLE_HTTP_SERVER_CONFIG_USER] ?? (extension_loaded('posix') ? posix_getpwuid(
+            posix_geteuid()
+        )['name'] : '-');
```
Comment (Rastusik) [inline src/Server/DefaultHttpServerConfiguration.php:212]: Why can't the user of the script be determined without having the posix extension?

**Review by Rastusik — COMMENTED** [review]
Hey @pistej ! Thank you very much for the contribution! Could you please also add tests for this new feature? Unit and "feature"? Thank you! 
It would be also ideal to have the correct user and group in the console output even without having to use the posix extension.

**Comment by Rastusik** [comment]
BTW not sure but maybe this could help? https://rohjay.one/unmasking-the-user-in-php-cli-scripts/

**Comment by pistej** [comment]
> BTW not sure but maybe this could help? https://rohjay.one/unmasking-the-user-in-php-cli-scripts/

not useful; didnt find user id in swoole server stats, neither in https://openswoole.com/docs/modules/swoole-http-request-server ... yes apache has some kind of this data, but apache :) ....

... will add test

**Comment by Rastusik** [comment]
> not useful; didnt find user id in swoole server stats, neither in https://openswoole.com/docs/modules/swoole-http-request-server ... yes apache has some kind of this data, but apache :) ....

so how about not using posix extension at all and just get the config value to the console output? seems to be easier and universal and there would be no need to install the extension. I think that if there was something wrong with the config values, the php script would exit with an error, wouldn't it?

**Comment by pistej** [comment]
> > not useful; didnt find user id in swoole server stats, neither in https://openswoole.com/docs/modules/swoole-http-request-server ... yes apache has some kind of this data, but apache :) ....
> 
> so how about not using posix extension at all and just get the config value to the console output? seems to be easier and universal and there would be no need to install the extension. I think that if there was something wrong with the config values, the php script would exit with an error, wouldn't it?

there is condition, and posix command is used only if it's installed and manual entry in config is not set, so think it;s safe
If user or group does not exist its used user who run server (command) (usually root), swoole runs fine, "just" warning about that

> btw are you aware that you don't have to wait for the pipeline to execute all the tests? you can run them locally ;) this info may save you some time ;) for now it's not very user friendly but I can explain to you how to do it

yes, I know that, :) I'm using my custom image with swoole a volume with code, ... but few lines about how to test, would be fine

**Comment by Rastusik** [comment]
I think that the posix part is not needed but if you really want it there, why not.🤷‍♂️ :)

I'll try to write Som dev info later but basically what needs to be done is to create a custom docker compose file with cli service from the original docker compose and add the needed docker build args and env variables combination (can be found in circle Ci config) and point to the docker file instead of a ready docker image

**Comment by pistej** [comment]
Soo, created 'feature' test, not sure what kind if unit test u want ?! Also fixed phpstan, but there will be still error with posix function description, (created fix in phpstan repo), but still its newer version than used here and we dont use baseline for this kind of errors, ...sooo

**Comment by Rastusik** [comment]
tests are ok, not sure what the phpstan problem is... maybe just suppress the problem, or maybe can you show me what do you mean?

**Comment by pistej** [comment]
I was referring to: https://phpstan.org/user-guide/baseline so instead of ignoring whole file in `excludePaths` or wirting `ignoreErrors` messages, u can generate baseline ... but this is nice to have

File: phpstan.neon [hunk 2]
```diff
@@ -19,4 +19,5 @@ parameters:
     ignoreErrors:
         # Put false positives here
         - '#PHPDoc tag @var for variable \$row contains unresolvable type#'
+        - "#Cannot access offset 'name' on array.*#"
```
Comment (Rastusik) [inline phpstan.neon:22]: I would prefer to use this instead: https://phpstan.org/user-guide/ignoring-errors ... no need to ignore the error everywhere and np need to use a baseline imho

Comment (pistej) [reply phpstan.neon]: my bad, didnt realize this option :), fixed

**Review by Rastusik — APPROVED** [review]

**Pull request merged**